src/firewall/firewallService.ts

```diff
@@ -0,0 +1,120 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { Account, CreateFirewallRuleRequest, HandleFirewallRuleRequest,
+    HandleFirewallRuleParams, HandleFirewallRuleResponse,
+    CreateFirewallRuleResponse,
+    AccountKey,
+    CreateFirewallRuleParams} from '../models/contracts/firewall/firewallRequest';
+import SqlToolsServiceClient from '../languageservice/serviceclient';
+import Constants = require('../constants/constants');
+import { Deferred } from '../protocol';
+import VscodeWrapper from '../controllers/vscodeWrapper';
+
+export class FirewallService {
+
+    private _isSignedIn: boolean = false;
+    private _session: any = undefined;
+    private _account: Account = undefined;
```

Comment:
It would be good to add the typings for the azure account extension so you don't have to have these any's all over the place [https://github.com/microsoft/vscode-azure-account/blob/master/src/azure-account.api.d.ts](https://github.com/microsoft/vscode-azure-account/blob/master/src/azure-account.api.d.ts)

Reply:
Is it worth adding the typings and add 3 new dependencies for the declarations? I could add this to interfaces, but some fields such as `Subscription` will stay as `any`

Reply:
Up to you in the end - but I know it's a huge annoyance for me working in a codebase that doesn't have typings since then I have to hunt around for what the types are and what properties they have. You're ignoring a huge part of typescript (the type safety) by doing this.